ops/framework.py

```diff
@@ -126,7 +126,13 @@ class EventBase:
     Inherit this and override 'snapshot' and 'restore' methods to build a custom event.
     """
 
-    def __init__(self, handle):
+    # gets patched in by `Framework.restore()`, if this event is being reemitted
+    # after being loaded from snapshot, or by `BoundEvent.emit()` if this
+    # event is being fired for the first time.
+    # TODO this is hard to debug, this should be refactored
+    framework = None
```

Comment:
Instead of doing this, you should be able to call `super().__init__(handle)`, as you do when defining a custom event.